src/components/GoodsSearchProvider.php

```diff
@@ -0,0 +1,413 @@
+use DotPlant\Store\models\price\ProductPrice;
+use yii\helpers\Url;
+use DotPlant\EntityStructure\models\BaseStructure;
+use DevGroup\DataStructure\helpers\PropertiesHelper;
+use yii\data\Pagination;
+use yii\data\Sort;
+use DotPlant\EntityStructure\helpers\PaginationHelper;
+
+class GoodsSearchProvider extends DataEntityProvider
+{
+    /**
+     * @var string the region key
+     */
+    public $regionKey = 'goodsSearchRegion';
+
+    /**
+     * @var string the material key
+     */
+    public $materialKey = 'goodsSearchMaterial';
+
+    /**
+     * @var string the current action route
+     */
+    public $actionRoute = ['/store/goods/search'];
+
+    /**
+     * @var string the block key
+     */
+    public $blockKey = 'goods';
+
+    /**
+     * @var string the pagination block key
+     */
+    public $paginationKey = 'pagination';
+
+    /**
+     * @var string the pagination block key
+     */
+    public $totalCountKey = 'totalCount';
+
+    /**
+     * @var string the categories block key
+     */
+    public $categoriesKey = 'categories';
+
+    /**
+     * @var string the image property key
+     */
+    public $imagePropertyKey = '';
+
+    /**
+     * @var int thumb width
+     */
+    public $thumbWidth = 262;
+
+    /**
+     * @var int thump height
+     */
+    public $thumbHeight = 175;
+
+    /**
+     * @var int goods per page
+     */
+    public $maxLimit = 12;
+
+
+    /**
+     * @var bool|int parent id
+     */
+    public $parentId = false;
+
+    /**
+     * @var string the pagination get parameter
+     */
+    public $paginationParameter = 'page';
+
+    /**
+     * @var string the search get parameter
+     */
+    public $searchParameter = 'search';
+
+    /**
+     * @var string the limit get parameter
+     */
+    public $limitParameter = 'limit';
+
+    /**
+     * @var string the limit get parameter
+     */
+    public $parentIdParameter = 'parent-id';
+
+    /**
+     * @var string the sort get parameter
+     */
+    public $sortParameter = 'sort';
+
+    /**
+     * @var array sort fields
+     */
+    public $sortFields = ['retail_price', 'id'];
+
+
+    public function pack()
+    {
+        return [
+            'class' => static::class,
+            'entities' => $this->entities,
+        ];
+    }
+
+    private function buildGood($good)
+    {
+        $language_id = \Yii::$app->multilingual->language_id;
+        $currency = CurrencyHelper::getUserCurrency();
+
+        if (!empty($this->imagePropertyKey)) {
+            $imageProperty = Property::findOne(['key' => $this->imagePropertyKey]);
+            $images = MediaHelper::getMediaData(
+                $good,
+                $imageProperty,
+                new GlideConfigurator($this->thumbWidth, $this->thumbHeight)
+            );
+            $image = reset($images);
+        }
+
+        ProductPrice::create($good);
+        $price = $good->getMinPrice(Price::TYPE_RETAIL, true, $currency->iso_code);
```

Comment:
use `Store::isRetail() ? Price::TYPE_RETAIL : Price::TYPE_WHOLESALE` instead of `Price::TYPE_RETAIL` in this case